File: skiCentar/UI/skicentar_desktop/lib/models/trail.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:skicentar_desktop/models/resort.dart';
import 'package:skicentar_desktop/models/trail_difficulty.dart';
import 'package:skicentar_desktop/models/trail_location.dart';

part 'trail.g.dart';

@JsonSerializable()

class Trail{
  int? id;
  String? name;
  bool? isFunctional;
  int? difficultyId;
  int? resortId;
  Resort? resort;
  double? length;
  TrailDifficulty? difficulty;
  List<TrailLocation>? trailLocations;
  
  Trail({this.id,this.name});

  factory Trail.fromJson(Map<String,dynamic> json) => _$TrailFromJson(json);

  Map<String,dynamic> toJson() => _$TrailToJson(this);
}

File: skiCentar/skiCentar.Model/Trail.cs
using System.Collections.Generic;

namespace skiCentar.Model
{
    public class Trail
    {
        public int Id { get; set; }

        public string Name { get; set; } = null!;

        public int? DifficultyId { get; set; }

        public decimal? Length { get; set; }

        public int? ResortId { get; set; }

        public bool? IsFunctional { get; set; }

        public virtual TrailDifficulty? Difficulty { get; set; }

        public virtual Resort? Resort { get; set; }

        public virtual ICollection<TrailLocation> TrailLocations { get; set; }

    }
}
